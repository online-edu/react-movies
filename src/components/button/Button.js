import React from 'react';
import PropTypes from 'prop-types';

/**
 * Button component.
 */
const Button = ({ aria, label, click }) => (
    <button
        type="button"
        onClick={click}
        aria-label={label}
        {...aria}
        className="btn btn-primary my-2"
    >
        {label}
    </button>
);
/**
 * Button default props.
 */
Button.defaultProps = {
    aria: {},
};
/**
 * Button props types.
 */
Button.propTypes = {
    /** Id to be assigned to the control. */
    aria: PropTypes.shape({ 'aria-controls': PropTypes.string }),
    /** Display value */
    label: PropTypes.string.isRequired,
    /** Gets called when the user clicks on button */
    click: PropTypes.func.isRequired,
};
export default Button;
