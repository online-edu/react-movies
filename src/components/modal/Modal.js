import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/Modal';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Poster from '../poster';

const MovieModal = ({ show, onHide, movie }) => (
    <Modal
        show={show}
        onHide={onHide}
        size="lg"
        aria-labelledby="movie-details"
        className="movie-modal"
        centered
    >
        <Modal.Header>
            <Modal.Title id="movie-details">{movie.title}</Modal.Title>
            <button
                type="button"
                aria-label="Close"
                onClick={onHide}
                className="btn btn-primary"
            >
                Close
            </button>
        </Modal.Header>
        <Modal.Body>
            <Row>
                <Col sm lg={4}>
                    <Poster name={movie.title} url={`${movie.poster_path}`} />
                </Col>
                <Col sm lg={8}>
                    <Row>
                        <Col md={12}>
                            <p className="h6 text-primary">Overview</p>
                            <p className="mb-0">
                                <em>{movie.overview}</em>
                            </p>
                        </Col>
                    </Row>
                    <div className="dropdown-divider" />
                    <dl className="row movie-modal__summary--list">
                        <dt className="col-sm-3 text-primary">Tagline</dt>
                        <dd className="col-sm-9">
                            {movie.tagline || 'Not available'}
                        </dd>
                        <dt className="col-sm-3 text-primary">Rating</dt>
                        <dd className="col-sm-9">{movie.vote_average}</dd>
                        <dt className="col-sm-3 text-primary">Status</dt>
                        <dd className="col-sm-9">{movie.status}</dd>
                        <dt className="col-sm-3 text-primary">Release Date</dt>
                        <dd className="col-sm-9">{movie.release_date}</dd>
                        <dt className="col-sm-3 text-primary">Language</dt>
                        <dd className="col-sm-9 text-uppercase">
                            {movie.original_language}
                        </dd>
                    </dl>
                </Col>
            </Row>
        </Modal.Body>
    </Modal>
);

MovieModal.propTypes = {
    show: PropTypes.bool.isRequired,
    onHide: PropTypes.func.isRequired,
    movie: PropTypes.shape({ title: PropTypes.string }).isRequired,
};

export default MovieModal;
