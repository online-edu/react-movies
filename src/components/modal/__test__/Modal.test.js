import React from 'react';
import { shallow, mount } from 'enzyme';
import Modal from '../Modal';
import Button from '../../button';

describe('<Modal />', () => {
    const onModalClose = jest.fn();
    const wrapper = (
        <Modal show title="Header" onHide={onModalClose}>
            <div>Content</div>
        </Modal>
    );

    it('renders correctly', () => {
        const component = shallow(wrapper);
        expect(component).toMatchSnapshot();
    });

    it('contains class name "movie-modal" ', () => {
        const component = shallow(wrapper);
        expect(component.find('.movie-modal')).toHaveLength(1);
    });

    it('renders a <Button /> component', () => {
        const component = shallow(wrapper);
        expect(component.find(Button)).toHaveLength(1);
    });

    it('allows us to set props', () => {
        const component = mount(wrapper);
        expect(component.props().show).toBe(true);
        component.setProps({ show: false });
        expect(component.props().show).toBe(false);
    });
});
