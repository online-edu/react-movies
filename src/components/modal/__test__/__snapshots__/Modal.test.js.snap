// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommonModal
    onHide={[MockFunction]}
    show={true}
    title="Header"
  >
    <div>
      Content
    </div>
  </CommonModal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "aria-labelledby": "movie-details",
      "centered": true,
      "children": Array [
        <ForwardRef(Bootstrap(ModalHeader))
          className="align-items-center"
        >
          <ForwardRef(Bootstrap(ModalTitle))
            id="movie-details"
          >
            Header
          </ForwardRef(Bootstrap(ModalTitle))>
          <Button
            aria={Object {}}
            click={[MockFunction]}
            label="Close"
          />
        </ForwardRef(Bootstrap(ModalHeader))>,
        <ForwardRef(Bootstrap(ModalBody))>
          <div>
            Content
          </div>
        </ForwardRef(Bootstrap(ModalBody))>,
      ],
      "className": "movie-modal",
      "onHide": [MockFunction],
      "show": true,
      "size": "lg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(Bootstrap(ModalTitle))
              id="movie-details"
            >
              Header
            </ForwardRef(Bootstrap(ModalTitle))>,
            <Button
              aria={Object {}}
              click={[MockFunction]}
              label="Close"
            />,
          ],
          "className": "align-items-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Header",
              "id": "movie-details",
            },
            "ref": null,
            "rendered": "Header",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria": Object {},
              "click": [MockFunction],
              "label": "Close",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <div>
            Content
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Content",
          },
          "ref": null,
          "rendered": "Content",
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "BACKDROP_TRANSITION_DURATION": 150,
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Dialog": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Footer": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "Header": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "TRANSITION_DURATION": 300,
      "Title": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
      "defaultProps": undefined,
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "aria-labelledby": "movie-details",
        "centered": true,
        "children": Array [
          <ForwardRef(Bootstrap(ModalHeader))
            className="align-items-center"
          >
            <ForwardRef(Bootstrap(ModalTitle))
              id="movie-details"
            >
              Header
            </ForwardRef(Bootstrap(ModalTitle))>
            <Button
              aria={Object {}}
              click={[MockFunction]}
              label="Close"
            />
          </ForwardRef(Bootstrap(ModalHeader))>,
          <ForwardRef(Bootstrap(ModalBody))>
            <div>
              Content
            </div>
          </ForwardRef(Bootstrap(ModalBody))>,
        ],
        "className": "movie-modal",
        "onHide": [MockFunction],
        "show": true,
        "size": "lg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(Bootstrap(ModalTitle))
                id="movie-details"
              >
                Header
              </ForwardRef(Bootstrap(ModalTitle))>,
              <Button
                aria={Object {}}
                click={[MockFunction]}
                label="Close"
              />,
            ],
            "className": "align-items-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Header",
                "id": "movie-details",
              },
              "ref": null,
              "rendered": "Header",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "aria": Object {},
                "click": [MockFunction],
                "label": "Close",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <div>
              Content
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Content",
            },
            "ref": null,
            "rendered": "Content",
            "type": "div",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "BACKDROP_TRANSITION_DURATION": 150,
        "Body": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Dialog": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Footer": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "Header": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "TRANSITION_DURATION": 300,
        "Title": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
